From d8b5da7bb16bfd3652e83200d851af3a21816469 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20Vr=C3=A1til?= <dvratil@redhat.com>
Date: Sun, 13 Sep 2015 15:05:08 +0200
Subject: [PATCH 24/31] Fix ItemSync merge type fallback

Always fallback to RID merge, even when GID merge is requsted, but no GID
is present.
---
 akonadi/itemsync.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/akonadi/itemsync.cpp b/akonadi/itemsync.cpp
index 9520637..e719e1a 100644
--- a/akonadi/itemsync.cpp
+++ b/akonadi/itemsync.cpp
@@ -130,10 +130,10 @@ void ItemSyncPrivate::createOrMerge(const Item &item)
     mPendingJobs++;
     ItemCreateJob *create = new ItemCreateJob(item, mSyncCollection, subjobParent());
     ItemCreateJob::MergeOptions merge = ItemCreateJob::Silent;
-    if (mMergeMode == ItemSync::RIDMerge) {
-        merge |= ItemCreateJob::RID;
-    } else if (mMergeMode == ItemSync::GIDMerge && !item.gid().isEmpty()) {
+    if (mMergeMode == ItemSync::GIDMerge && !item.gid().isEmpty()) {
         merge |= ItemCreateJob::GID;
+    } else {
+        merge |= ItemCreateJob::RID;
     }
     create->setMerge(merge);
     q->connect(create, SIGNAL(result(KJob*)), q, SLOT(slotLocalChangeDone(KJob*)));
-- 
2.5.0

